lib/parser.js

```diff
@@ -0,0 +1,115 @@
+'use strict'
+const path = require('path')
+
+module.exports = class Parser {
+
+  static commandParser (opts) {
+    const command = opts.command
+    if (!command) return null
+    if (Array.isArray(command)) {
+      return command.join(' ')
+    }
+    return command
+  }
+
+  static buildDockerfilePathParser (opts) {
+    const build = opts.build
+    const warnings = opts.warnings
+    let dockerFileName = 'Dockerfile'
+    if (!build) return null
+
+    if (build.args) {
+      // NOTE: Maybe pass these to the ENVs?
+      warnings.add('The `args` argument is not supported for builds. These args will not be passed to the build.', { args: build.args })
+    }
+
+    let buildPath
+    if (typeof build === 'string') {
+      buildPath = build
+    } else {
+      buildPath = build.context
+      if (build.dockerfile) {
+        dockerFileName = build.dockerfile
+      }
+    }
+    return path.resolve('/', buildPath, dockerFileName)
+  }
+
+  static portsParser (opts) {
+    const ports = opts.ports || []
+    const expose = opts.expose || []
+    const warnings = opts.warnings
+    const allPorts = ports.concat(expose)
+    return allPorts.map(port => {
+      if (port.match(/^\d+$/)) {
+        return parseInt(port, 10)
+      }
+      if (port.match(/^\d+:\d+$/)) {
+        const ports = port.match(/^(\d+):(\d+)$/)
+        if (ports[1] === ports[2]) {
+          return parseInt(ports[1], 10)
+        }
+        warnings.add('An invalid port mapping was added and will be ignored', { ports: [ports[1], ports[2]] })
+        return null
+      }
+      warnings.add('An invalid port was added and will be ignored', { port })
+      return null
+    })
+    .filter(x => !!x)
+  }
+
+  static envParser (opts) {
+    const env = opts.env
+    return env
+  }
+
+  static filesParser (opts) {
+    const image = opts.image
+    if (image) {
+      return {
+        '/Dockerfile': {
+          body:
+`# Image automatically created from docker-compose file
+FROM ${image}`
+        }
+      }
+    }
+    return null
+  }
+
+  static nameParser (opts) {
```

Comment:
This seems to be a reoccurring pattern throughout the codebase. It seems you prefer wording functions for what they *are* rather then what they *do*. Of course this is a stylistic thing, but I learned from the book *clean code* 

can't find the link to that part right now but summary:
```
Class Names should usually be Nouns.
Method names should usually be Verbs.
```

Do with this info what you will.